src/blur.js

```diff
@@ -0,0 +1,64 @@
+function blurTransfer(V, r, n, vertical) {
+  const [source, target] = V,
+    m = Math.floor(source.length / n),
+    w = (r << 1) + 1,
```

Comment:
Is this an implementation of a well known formula/algorithm? Anywhere I could read up on this math? I'm finding it a bit difficult to follow. Thanks!

Reply:
Reading up on left shift now... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#Left_shift

Reply:
Looks like this is equivalent to `w = r * 2 + 1`? Is the `<<` a performance optimization?

Reply:
It also performs coercing the value to integer. Note that this was in the original code; it doesn't bring much in terms of performance since it's only called once.